ballerina/field.bal

```diff
@@ -97,6 +107,36 @@ public class Field {
         return self.fieldValue;
     }
 
+    isolated function getFieldObjects(parser:SelectionNode selectionNode, __Type 'type) returns Field[] {
+        string[] currentPath = self.path.clone().'map((item) => item is int ? "@" : item);
+        string[] unwrappedPath = getUnwrappedPath('type);
+        __Type unwrappedType = getOfType('type);
+        
+        if unwrappedType.kind != OBJECT && unwrappedType.kind != INTERFACE {
+            return [];
+        }
+
+        Field[] result = [];
+        foreach parser:SelectionNode selection in selectionNode.getSelections() {
+            if selection is parser:FieldNode {
+                foreach __Field 'field in <__Field[]>unwrappedType.fields {
```

Comment:
We can remove the cast here by checking `unwrappedType.fields` value at line 115 instead of checking the kind